websocket/src/websocket.cpp

```diff
@@ -53,6 +54,7 @@ const char* ResultToString(Result err)
 const char* StateToString(State err)
 {
     switch(err) {
+        STRING_CASE(STATE_CONNECT);
```

Comment:
I think these two names are too confusing.
Once "Connect" is done, I'd expect is it connected. But then we come to "Connecting"?

Reply:
Yes, it doesn't really sit good with me either. With WSLAY a sequence of events happen:

* CONNECT
* HANDSHAKE_WRITE
* HANDSHAKE_READ
* CONNECTED

With Emscripten it's a bit different:

* CONNECT
* This kicks off an asynchronous connection attempt with Emscripten where the result is reported in a callback function
* CONNECTING (while waiting for the callback)
* CONNECTED (when receiving a successful callback)

Note that the state is checked in OnUpdate so we need to transition from CONNECT to something else while waiting for the asynchronous response for the Emscripten socket.

Reply:
"CONNECTING" for emscripten could also be seen as "handshake read"?